internal/controller/bucket/helpers.go

```diff
@@ -57,12 +57,10 @@ func isBucketAvailableFromStatus(bucket *v1alpha1.Bucket, backendClients map[str
 			continue
 		}
 
-		if _, ok := bucket.Status.AtProvider.Backends[backendName]; !ok {
+		if backend, ok := bucket.Status.AtProvider.Backends[backendName]; !ok || backend == nil {
```

Comment:
IMO: if we check whether backend is nil or not, I think we don't need also use `ok`.
Because in those cases backend is nil:
1. backendName is not assigned in the map
2. Previously backendName is assigned, but then the value is set as nil
3. backendName is explicitly set as nil